tests/lib/structures/Josh.test.ts

```diff
@@ -21,4 +137,1083 @@ describe('Josh', () => {
+      });
+
+      test('GIVEN josh w/o data THEN returns data AND sets value at key and path', async () => {
+        const hasBefore = await josh[Method.Has]('test:set', ['path']);
+
+        expect(hasBefore).toBe(false);
+
+        const set = await josh[Method.Set]('test:set', 'value', 'path');
+
+        expect(set).toBeInstanceOf(Josh);
+
+        const hasAfter = await josh[Method.Has]('test:set', ['path']);
+
+        expect(hasAfter).toBe(true);
+      });
+    });
+
+    describe(Method.SetMany, () => {
+      test('GIVEN josh w/o data THEN returns data AND sets value at key', async () => {
+        const hasBefore = await josh[Method.Has]('test:setMany');
+
+        expect(hasBefore).toBe(false);
+
+        const setMany = await josh[Method.SetMany]([
+          [{ key: 'test:setMany', path: [] }, 'value'],
+          [{ key: 'test:setMany2' }, 'value']
+        ]);
+
+        expect(setMany).toBeInstanceOf(Josh);
+
+        const entries = await josh[Method.Entries]();
+
+        expect(entries).toEqual({ 'test:setMany': 'value', 'test:setMany2': 'value' });
+      });
+
+      test('GIVEN josh w/ data THEN returns data AND does not set value at key', async () => {
+        await josh[Method.Set]('test:setMany', 'value');
+
+        const setMany = await josh[Method.SetMany]([[{ key: 'test:setMany', path: [] }, 'value-overwritten']], false);
+
+        expect(setMany).toBeInstanceOf(Josh);
+
+        const entries = await josh[Method.Entries]();
+
+        expect(entries).toEqual({ 'test:setMany': 'value' });
+      });
+    });
+
+    describe(Method.Size, () => {
+      test('GIVEN josh w/o data THEN returns 0', async () => {
+        const size = await josh[Method.Size]();
+
+        expect(size).toBe(0);
+      });
+
+      test('GIVEN josh w/ data THEN returns 1)', async () => {
+        await josh[Method.Set]('test:size', 'value');
+
+        const size = await josh[Method.Size]();
+
+        expect(size).toBe(1);
+      });
+    });
+
+    describe(Method.Some, () => {
+      describe(Payload.Type.Hook, () => {
+        test('GIVEN josh w/o data THEN returns false', async () => {
+          const some = await josh[Method.Some]((value) => value === 'value');
+
+          expect(some).toBe(false);
+        });
+
+        test('GIVEN josh w/ data w/o value THEN rejects', async () => {
+          const some = josh[Method.Some]('path', undefined);
+
+          await expect(some).rejects.toBeInstanceOf(JoshError);
+        });
+
+        test('GIVEN josh w/ data w/ invalid primitive THEN rejects', async () => {
+          const some = josh[Method.Some]('path', null);
+
+          await expect(some).rejects.toBeInstanceOf(JoshError);
+        });
+
+        // test('GIVEN josh w/ data THEN returns true', async () => {
+        //   await josh[Method.Set]('test:some', 'value');
+
+        //   const some = await josh[Method.Some]([], 'value');
+
+        //   expect(some).toBe(true);
+        // }); // For Nova
```

Comment:
what's up with this as well

Reply:
These error with the getProperty issue where path is empty, some and every by value throws a missing data error

Reply:
Ok, can we create an issue for this on the utilities repository so a test can be added and then adapted in the providers repository when updating the dependency.